# Notebook for running all the examples in the examples folder

This notebook also serves as a test of the ras_commander package using the HEC-RAS Project Examples provided by HEC.

Example project names can be found in the example_projects.csv file after the examples are run.

To use a different example, change the project name in the examples and run them directly or run this notebook.  


The notebook format provides a convenient way to run all the examples to ensure that no features are broken after code revisions.  


-----

This table is a valuable reference for formulating examples and anticipating script behavior.

| Project           | Type     | Existing Numbers | Next Available |
|-------------------|----------|-------------------|----------------|
| Balde Eagle Creek | Plan     | 01, 02            | 03             |
|                   | Flow     | 01, 02            | 03             |
|                   | Unsteady | 02                | 01, 03         |
|                   | Geometry | 01                | 02             |
| Muncie            | Plan     | 01, 03, 04        | 02, 05         |
|                   | Flow     | 01                | 02             |
|                   | Unsteady | 01                | 02             |
|                   | Geometry | 01, 02, 04        | 03, 05         |

Notes:
1. For Balde Eagle Creek's Unsteady entries, 01 is available before 03. The library should use the lowest available number when creating new entries.
2. For Muncie's Plan entries, both 02 and 05 are available. The library should use 02 first.
3. For Muncie's Geometry entries, 03 is available before 05.

When writing examples or scripts:
1. Use existing numbers when referencing current plans, flows, unsteady setups, or geometries.
2. When creating new entries (e.g., cloning), use the "Next Available" number and then increment for subsequent creations.
3. After operations that create new entries, verify that the ras object has been updated to reflect these changes.

This approach will help ensure that examples run as expected and that we're always working with valid, up-to-date data in our scripts.

-----

# RAS-Commander (ras_commander) library examples

### The Examples Below Have Been Tested and Verified.

-----

In [1]:
#!pip install pandas pathlib requests psutil h5py matplotlib scipy

In [2]:
# Example 1: Import and execute 01_project_initialization.py
print("Executing 01_project_initialization.py:")
%run 01_project_initialization.py
print("\n")

Executing 01_project_initialization.py:


2024-11-12 13:52:32,499 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 13:52:32,503 - ras_commander.RasExamples - WARNING - No existing example projects zip file found.
2024-11-12 13:52:32,504 - ras_commander.RasExamples - INFO - No example projects zip file found. Downloading...
2024-11-12 13:52:32,505 - ras_commander.RasExamples - INFO - Getting example projects for version 6.6
2024-11-12 13:52:32,507 - ras_commander.RasExamples - INFO - Downloading HEC-RAS Example Projects from https://github.com/HydrologicEngineeringCenter/hec-downloads/releases/download/1.0.33/Example_Projects_6_6.zip. 
The file is over 400 MB, so it may take a few minutes to download....
2024-11-12 13:52:38,759 - ras_commander.RasExamples - INFO - Downloaded to c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 13:52:38,760 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip

Example Set 1: Using the default global 'ras' object
-----------------------------------------------------
Step 1: Initializing with global RAS instance

Global RAS Instance (Bald Eagle Creek) Project Data:
Project Name: BaldEagle
Project Folder: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek
PRJ File: C:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.prj
HEC-RAS Executable Path: C:\Program Files (x86)\HEC\HEC-RAS\6.5\Ras.exe

Plan Files DataFrame:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version  ..

ValueError: No HEC-RAS project file found in c:\GH\ras-commander\examples\example_projects\BaldEagleCrkMulti2D

-----

In [3]:
import time

# wait 10 seconds
time.sleep(10)

In [4]:
# Example 2: Import and execute 02_plan_operations.py
print("Executing 02_plan_operations.py:")
%run 02_plan_operations.py
print("\n")

2024-11-12 13:52:50,658 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 13:52:50,659 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 13:52:50,661 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 13:52:50,663 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 13:52:50,664 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 13:52:50,665 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 13:52:50,666 - ras_commander.RasExamples - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-11-12 13:52:50,676 - ras_commander.RasExamples - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-11-12 13:52:50,754 - ras_

Executing 02_plan_operations.py:
Initial plan files:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version  ...  \
0            1HOUR  Unsteady with Bridges and Dam            5.00  ...   
1              NaN                Steady Flow Run             NaN  ...   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN        SteadyRun   

                   Simulation Date UNET D1 Cores UNET Use Existing IB Tables

2024-11-12 13:52:50,838 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:52:50,842 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 13:52:50,843 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 13:52:50,846 - ras_commander.RasPrj - INFO - Found 1 geometry entries
2024-11-12 13:52:50,846 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 13:52:50,847 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 13:52:50,848 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:52:50,848 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:52:50,850 - ras_commander.RasPrj - INFO - Finished get_flow_entries
2024-11-12 13:52:50,851 - ras_commander.RasPrj - INFO - Calling get_unsteady_entries
2024-11-12 13:52:50,851 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:52:50,852 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13

New plan created: 03
Updated plan files:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   
2          03  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   
2                 2MIN      dss       u02                     2       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   
2            1HOUR  Unsteady with Bridges and Dam            5.00        1   

  Run PostProcess Run Sediment Run UNet Run WQNet   Short Identi

2024-11-12 13:52:51,042 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 13:52:51,043 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:52:51,043 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:52:51,048 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 13:52:51,050 - ras_commander.RasPlan - INFO - Finished get_plan_path
2024-11-12 13:52:51,050 - ras_commander.RasUtils - INFO - Calling update_file
2024-11-12 13:52:51,053 - ras_commander.RasUtils - INFO - Successfully updated file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p03
2024-11-12 13:52:51,054 - ras_commander.RasUtils - INFO - Finished update_file
2024-11-12 13:52:51,054 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 13:52:51,055 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:52:51,055 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:52:51,059 - 

Updated unsteady flow for plan 03

Step 6: Setting the number of cores for a plan
Updated number of cores for plan 03

Step 7: Setting geometry preprocessor options
Updated geometry preprocessor options for plan 03

Step 8: Updating simulation parameters
Updated simulation parameters

Step 10: Computing the cloned plan


2024-11-12 13:55:50,385 - ras_commander.RasCmdr - INFO - HEC-RAS execution completed for plan: 03
2024-11-12 13:55:50,385 - ras_commander.RasCmdr - INFO - Total run time for plan 03: 179.19 seconds
2024-11-12 13:55:50,388 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 13:55:50,389 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:55:50,390 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:55:50,395 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 13:55:50,396 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 13:55:50,400 - ras_commander.RasPrj - INFO - Found 2 geometry entries
2024-11-12 13:55:50,401 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 13:55:50,402 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 13:55:50,403 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:55:50,404 - ras_commander.RasPrj - INFO - Finished ch

Computing plan 03
Plan 03 computed successfully

Step 11: Retrieving HDF entries for the cloned plan
HDF entries for the cloned plan:
  plan_number                                          full_path  \
2          03  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
2                 1MIN      dss       u02                     2       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
2            30MIN  Unsteady with Bridges and Dam            5.00        1   

  Run PostProcess Run Sediment Run UNet Run WQNet   Short Identifier  \
2               1            0        1         0  UnsteadyFlow_copy   

                 Simulation Date UNET D1 Cores UNET Use Existing IB Tables  \
2  01JAN2023,0000,05JAN2023,2359             2                          -1   

                                    HDF_Results_Path  
2  c:\GH\ras-commander\examples\example_projects\...  

Curr

-----

In [5]:
# wait 10 seconds
time.sleep(10)

In [6]:
# Example 3: Import and execute 03_geometry_operations.py
print("Executing 03_geometry_operations.py:")
%run 03_geometry_operations.py
print("\n")

2024-11-12 13:56:00,474 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 13:56:00,475 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 13:56:00,478 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 13:56:00,482 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 13:56:00,483 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 13:56:00,485 - ras_commander.RasExamples - INFO - Extracting project 'Muncie'
2024-11-12 13:56:00,486 - ras_commander.RasExamples - INFO - Project 'Muncie' already exists. Deleting existing folder...
2024-11-12 13:56:00,536 - ras_commander.RasExamples - INFO - Existing folder for project 'Muncie' has been deleted.


Executing 03_geometry_operations.py:


2024-11-12 13:56:01,191 - ras_commander.RasExamples - INFO - Successfully extracted project 'Muncie' to c:\GH\ras-commander\examples\example_projects\Muncie
2024-11-12 13:56:01,192 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-11-12 13:56:01,194 - ras_commander.RasPrj - INFO - Calling init_ras_project
2024-11-12 13:56:01,195 - ras_commander.RasPrj - INFO - Calling get_ras_exe
2024-11-12 13:56:01,197 - ras_commander.RasPrj - INFO - HEC-RAS executable found at default path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-11-12 13:56:01,198 - ras_commander.RasPrj - INFO - Finished get_ras_exe
2024-11-12 13:56:01,199 - ras_commander.RasPrj - INFO - Initializing global 'ras' object via init_ras_project function.
2024-11-12 13:56:01,200 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 13:56:01,201 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 13:56:01,205 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-1

Initial plan files:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          03  c:\GH\ras-commander\examples\example_projects\...   
2          04  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                15SEC      dss       u01                     1       g01   
1                10SEC      dss       u01                     1       g02   
2                10SEC      dss       u01                     1       g04   

  Mapping Interval                                Plan Title Program Version  \
0             5MIN                  Unsteady Multi  9-SA run            5.00   
1             5MIN            Unsteady Run with 2D 50ft Grid            5.10   
2             5MIN  Unsteady Run with 2D 50ft User n Value R            5.10   

   ... Run WQNet     Short Identifier                Simulation Date  \
0  ..

2024-11-12 13:56:01,426 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:56:01,427 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 13:56:01,428 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:56:01,429 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:56:01,439 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 13:56:01,440 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 13:56:01,447 - ras_commander.RasPrj - INFO - Found 4 geometry entries
2024-11-12 13:56:01,448 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 13:56:01,450 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 13:56:01,451 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:56:01,455 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:56:01,460 - ras_commander.RasPrj - INFO - Finished get_flow_entries
2024-11-12 13:56:

New geometry 03 assigned to plan 02
Updated geometry files:
  geom_file geom_number                                          full_path  \
0       g01          01  c:\GH\ras-commander\examples\example_projects\...   
1       g02          02  c:\GH\ras-commander\examples\example_projects\...   
2       g04          04  c:\GH\ras-commander\examples\example_projects\...   
3       g03          03  c:\GH\ras-commander\examples\example_projects\...   

                                            hdf_path  
0  c:\GH\ras-commander\examples\example_projects\...  
1  c:\GH\ras-commander\examples\example_projects\...  
2  c:\GH\ras-commander\examples\example_projects\...  
3  c:\GH\ras-commander\examples\example_projects\...  

Step 3: Clearing geometry preprocessor files for the cloned plan
Cleared geometry preprocessor files for plan 02

Step 4: Clearing geometry preprocessor files for all plans
Cleared geometry preprocessor files for all plans

Step 5: Updated plan information
  plan_number   

2024-11-12 13:56:38,089 - ras_commander.RasCmdr - INFO - HEC-RAS execution completed for plan: 02
2024-11-12 13:56:38,090 - ras_commander.RasCmdr - INFO - Total run time for plan 02: 36.45 seconds
2024-11-12 13:56:38,091 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 13:56:38,092 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:56:38,092 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 13:56:38,098 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 13:56:38,099 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 13:56:38,103 - ras_commander.RasPrj - INFO - Found 4 geometry entries
2024-11-12 13:56:38,104 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 13:56:38,105 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 13:56:38,106 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 13:56:38,107 - ras_commander.RasPrj - INFO - Finished che

Computing plan 02
Plan 02 computed successfully

Step 7: Getting results paths
Results for plan 02 are located at: c:\GH\ras-commander\examples\example_projects\Muncie\Muncie.p02.hdf
No results found for plan 01




-----

In [7]:
# wait 10 seconds
time.sleep(10)

In [8]:
# Example 4: Import and execute 04_unsteady_flow_operations.py
print("Executing 04_unsteady_flow_operations.py:")
%run 04_unsteady_flow_operations.py
print("\n")

2024-11-12 13:56:48,168 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 13:56:48,169 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 13:56:48,170 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 13:56:48,173 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 13:56:48,174 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 13:56:48,174 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 13:56:48,175 - ras_commander.RasExamples - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-11-12 13:56:48,193 - ras_commander.RasExamples - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-11-12 13:56:48,257 - ras_

Executing 04_unsteady_flow_operations.py:
Initial unsteady flow files:
  unsteady_number                                          full_path  \
0              02  c:\GH\ras-commander\examples\example_projects\...   

          Flow Title Program Version Use Restart Precipitation Mode  \
0  Flow Hydrograph 2            6.30           0            Disable   

        Wind Mode Met BC=Precipitation|Mode Met BC=Evapotranspiration|Mode  \
0  No Wind Forces                      None                           None   

  Met BC=Precipitation|Expanded View Met BC=Precipitation|Constant Units  \
0                                  0                               mm/hr   

  Met BC=Precipitation|Gridded Source  
0                                 DSS  

Step 1: Extracting boundary conditions and tables
Unsteady file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.u02
Extracted boundary conditions and tables
Step 2: Printing boundaries and tables

Boundaries and Tablesin bo

-----

In [9]:
# wait 10 seconds
time.sleep(10)

In [10]:
# Example 5: Import and execute 05_utility_functions.py
print("Executing 05_utility_functions.py:")
%run 05_utility_functions.py
print("\n")

2024-11-12 13:56:58,342 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 13:56:58,343 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 13:56:58,345 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 13:56:58,347 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 13:56:58,348 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 13:56:58,348 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 13:56:58,349 - ras_commander.RasExamples - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-11-12 13:56:58,354 - ras_commander.RasExamples - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-11-12 13:56:58,407 - ras_

Executing 05_utility_functions.py:
Example 1: Getting plan path
Path for plan 01 is: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p01

Example 2: Getting geometry path
Path for geometry 01 is: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.g01

Example 3: Getting unsteady flow path
Path for unsteady flow 01 is: None

Example 4: Getting project name
Project name: BaldEagle




-----

In [11]:
# wait 10 seconds
time.sleep(10)

In [12]:
# Example 6: Import and execute 06_single_plan_execution.py
print("Executing 06_single_plan_execution.py:")
%run 06_single_plan_execution.py
print("\n")

Executing 06_single_plan_execution.py:


PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'c:\\GH\\ras-commander\\examples\\example_projects\\BaldEagleCrkMulti2D'

-----

In [13]:
# wait 10 seconds
time.sleep(10)

In [14]:
# Example 7: Import and execute 07_sequential_plan_execution.py
print("Executing 07_sequential_plan_execution.py:")
%run 07_sequential_plan_execution.py
print("\n")

Executing 07_sequential_plan_execution.py:
Removing existing folder: C:\GH\ras-commander\examples\example_projects


PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\GH\\ras-commander\\examples\\example_projects\\BaldEagleCrkMulti2D'

-----

In [15]:
# wait 10 seconds
time.sleep(10)

In [16]:
# Example 8: Import and execute 08_parallel_execution.py
print("Executing 08_parallel_execution.py:")
%run 08_parallel_execution.py
print("\n")

2024-11-12 13:57:28,705 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 13:57:28,706 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 13:57:28,707 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 13:57:28,710 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 13:57:28,710 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 13:57:28,711 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 13:57:28,770 - ras_commander.RasExamples - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-11-12 13:57:28,771 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-1

Executing 08_parallel_execution.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN        SteadyRun   

                   Simulation Date UNET D1 Cores UNET Use Existi

2024-11-12 13:57:28,894 - ras_commander.RasPrj - INFO - Calling init_ras_project
2024-11-12 13:57:28,895 - ras_commander.RasPrj - INFO - Calling get_ras_exe
2024-11-12 13:57:28,896 - ras_commander.RasPrj - INFO - HEC-RAS executable found at specified path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-11-12 13:57:28,896 - ras_commander.RasPrj - INFO - Finished get_ras_exe
2024-11-12 13:57:28,897 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 13:57:28,898 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 13:57:28,901 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 13:57:28,902 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 13:57:28,908 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 13:57:28,911 - ras_commander.RasPrj - INFO - Found 1 geometry entries
2024-11-12 13:57:28,912 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 13:57:28,915 - ras_commander.RasPrj - INFO - Finish

Parallel execution of all plans results:
Plan 01: Successful
Plan 02: Successful

Plan DataFrame after parallel execution of all plans:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN   

2024-11-12 14:00:28,514 - ras_commander.RasCmdr - INFO - Created worker folder: c:\GH\ras-commander\examples\example_projects\compute_test_parallel_specific [Worker 1]
2024-11-12 14:00:28,515 - ras_commander.RasPrj - INFO - Calling init_ras_project
2024-11-12 14:00:28,516 - ras_commander.RasPrj - INFO - Calling get_ras_exe
2024-11-12 14:00:28,517 - ras_commander.RasPrj - INFO - HEC-RAS executable found at specified path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-11-12 14:00:28,517 - ras_commander.RasPrj - INFO - Finished get_ras_exe
2024-11-12 14:00:28,517 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:00:28,518 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:00:28,521 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:00:28,522 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 14:00:28,527 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:00:28,530 - ras_commander.RasPrj - INFO - F

Parallel execution of specific plans results:
Plan 01: Successful
Plan 02: Successful

Example 3: Parallel execution with dynamic max_workers


2024-11-12 14:03:08,615 - ras_commander.RasCmdr - INFO - Created worker folder: c:\GH\ras-commander\examples\example_projects\compute_test_parallel_dynamic [Worker 1]
2024-11-12 14:03:08,616 - ras_commander.RasPrj - INFO - Calling init_ras_project
2024-11-12 14:03:08,618 - ras_commander.RasPrj - INFO - Calling get_ras_exe
2024-11-12 14:03:08,619 - ras_commander.RasPrj - INFO - HEC-RAS executable found at specified path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-11-12 14:03:08,620 - ras_commander.RasPrj - INFO - Finished get_ras_exe
2024-11-12 14:03:08,620 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:03:08,621 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:03:08,625 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:03:08,625 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 14:03:08,633 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:03:08,636 - ras_commander.RasPrj - INFO - Fo

Parallel execution with 3 workers and 2 cores per worker:
Plan 01: Successful
Plan 02: Successful

Results paths for dynamic execution:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN   

-----

In [17]:
# wait 10 seconds
time.sleep(10)

In [18]:
# Example 9: Import and execute 09_specifying_plans.py
print("Executing 09_specifying_plans.py:")
%run 09_specifying_plans.py
print("\n")

Executing 09_specifying_plans.py:
Removing existing folder: C:\GH\ras-commander\examples\example_projects


PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'C:\\GH\\ras-commander\\examples\\example_projects\\BaldEagleCrkMulti2D'

-----

In [19]:
# wait 10 seconds
time.sleep(10)

In [20]:
print("Executing 10_arguments_for_compute.py:")
%run 10_arguments_for_compute.py
print("\n")

2024-11-12 14:06:32,706 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 14:06:32,707 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 14:06:32,709 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 14:06:32,712 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 14:06:32,712 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 14:06:32,713 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 14:06:32,780 - ras_commander.RasExamples - INFO - Successfully extracted project 'Balde Eagle Creek' to c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek
2024-11-12 14:06:32,781 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-1

Executing 10_arguments_for_compute.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN        SteadyRun   

                   Simulation Date UNET D1 Cores UNET Use Exi

2024-11-12 14:06:32,877 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:06:32,878 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 14:06:32,884 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:06:32,887 - ras_commander.RasPrj - INFO - Found 1 geometry entries
2024-11-12 14:06:32,888 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 14:06:32,892 - ras_commander.RasPrj - INFO - Finished _load_project_data
2024-11-12 14:06:32,892 - ras_commander.RasPrj - INFO - Calling get_boundary_conditions
2024-11-12 14:06:32,898 - ras_commander.RasPrj - INFO - Finished get_boundary_conditions
2024-11-12 14:06:32,898 - ras_commander.RasPrj - INFO - Initialization complete for project: BaldEagle
2024-11-12 14:06:32,899 - ras_commander.RasPrj - INFO - Plan entries: 2, Flow entries: 2, Unsteady entries: 1, Geometry entries: 1, Boundary conditions: 3
2024-11-12 14:06:32,901 - ras_commander.RasPrj - INFO - Geometry HDF files f

Plan 01 execution: Successful


2024-11-12 14:10:38,787 - ras_commander.RasCmdr - INFO - HEC-RAS execution completed for plan: 02
2024-11-12 14:10:38,787 - ras_commander.RasCmdr - INFO - Total run time for plan 02: 9.09 seconds
2024-11-12 14:10:38,788 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 14:10:38,789 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:10:38,789 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:10:38,794 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:10:38,794 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:10:38,797 - ras_commander.RasPrj - INFO - Found 1 geometry entries
2024-11-12 14:10:38,798 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 14:10:38,798 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 14:10:38,799 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:10:38,799 - ras_commander.RasPrj - INFO - Finished chec

Plan 02 execution: Successful
Sequential execution completed

Example 2: Parallel execution with various arguments


2024-11-12 14:10:39,016 - ras_commander.RasPrj - INFO - Finished _load_project_data
2024-11-12 14:10:39,016 - ras_commander.RasPrj - INFO - Calling get_boundary_conditions
2024-11-12 14:10:39,021 - ras_commander.RasPrj - INFO - Finished get_boundary_conditions
2024-11-12 14:10:39,021 - ras_commander.RasPrj - INFO - Initialization complete for project: BaldEagle
2024-11-12 14:10:39,022 - ras_commander.RasPrj - INFO - Plan entries: 2, Flow entries: 2, Unsteady entries: 1, Geometry entries: 1, Boundary conditions: 3
2024-11-12 14:10:39,023 - ras_commander.RasPrj - INFO - Geometry HDF files found: 2
2024-11-12 14:10:39,024 - ras_commander.RasPrj - INFO - Finished initialize
2024-11-12 14:10:39,024 - ras_commander.RasPrj - INFO - Project initialized. ras_instance project folder: c:\GH\ras-commander\examples\example_projects\parallel_results [Worker 2]
2024-11-12 14:10:39,025 - ras_commander.RasPrj - INFO - Finished init_ras_project
2024-11-12 14:10:39,027 - ras_commander.RasCmdr - INFO - Ca

Parallel execution results:
Plan 01: Successful
Plan 02: Successful

Example 3: Single plan execution with specific arguments


2024-11-12 14:13:30,613 - ras_commander.RasCmdr - INFO - HEC-RAS execution completed for plan: 02
2024-11-12 14:13:30,614 - ras_commander.RasCmdr - INFO - Total run time for plan 02: 9.18 seconds
2024-11-12 14:13:30,615 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 14:13:30,616 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:13:30,617 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:13:30,622 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:13:30,622 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:13:30,626 - ras_commander.RasPrj - INFO - Found 1 geometry entries
2024-11-12 14:13:30,627 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 14:13:30,629 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 14:13:30,630 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:13:30,630 - ras_commander.RasPrj - INFO - Finished chec

Single plan execution: Successful




In [21]:
# wait 10 seconds
time.sleep(10)

In [22]:
print("Executing 12_plan_set_execution.py:")
%run 12_plan_set_execution.py
print("\n")

2024-11-12 14:13:40,665 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 14:13:40,665 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 14:13:40,666 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 14:13:40,669 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 14:13:40,669 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 14:13:40,669 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 14:13:40,670 - ras_commander.RasExamples - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-11-12 14:13:40,675 - ras_commander.RasExamples - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-11-12 14:13:40,724 - ras_

Executing 12_plan_set_execution.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN        SteadyRun   

                   Simulation Date UNET D1 Cores UNET Use Existi

2024-11-12 14:13:40,848 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:13:40,848 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:13:40,851 - ras_commander.RasPrj - INFO - Found 2 geometry entries
2024-11-12 14:13:40,852 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 14:13:40,852 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 14:13:40,853 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:13:40,853 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:13:40,855 - ras_commander.RasPrj - INFO - Finished get_flow_entries
2024-11-12 14:13:40,856 - ras_commander.RasPrj - INFO - Calling get_unsteady_entries
2024-11-12 14:13:40,856 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:13:40,857 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:13:40,859 - ras_commander.RasPrj - INFO - Finished get_unsteady_entries
2024-11-12

Created plan set:
  plan_number geom_number
0          03          02
1          04          03
2          05          04
3          06          05
4          07          06

Executing plan set in parallel


2024-11-12 14:13:42,072 - ras_commander.RasCmdr - INFO - Created worker folder: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [Worker 2]
2024-11-12 14:13:42,073 - ras_commander.RasPrj - INFO - Calling init_ras_project
2024-11-12 14:13:42,074 - ras_commander.RasPrj - INFO - Calling get_ras_exe
2024-11-12 14:13:42,075 - ras_commander.RasPrj - INFO - HEC-RAS executable found at specified path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-11-12 14:13:42,076 - ras_commander.RasPrj - INFO - Finished get_ras_exe
2024-11-12 14:13:42,077 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:13:42,077 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:13:42,081 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:13:42,082 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 14:13:42,092 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:13:42,095 - ras_commander.RasPrj - INFO - Found 6 geomet


Plan set execution results:
  plan_number geom_number  execution_success
0          03          02               True
1          04          03               True
2          05          04               True
3          06          05               True
4          07          06               True




In [23]:
# wait 10 seconds
time.sleep(10)

In [24]:
print("Executing 13_multiple_project_operations.py:")
%run 13_multiple_project_operations.py
print("\n")

2024-11-12 14:18:45,804 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 14:18:45,805 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 14:18:45,806 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 14:18:45,809 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 14:18:45,809 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 14:18:45,810 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 14:18:45,811 - ras_commander.RasExamples - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-11-12 14:18:45,823 - ras_commander.RasExamples - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-11-12 14:18:45,900 - ras_

Executing 13_multiple_project_operations.py:


2024-11-12 14:18:46,373 - ras_commander.RasExamples - INFO - Successfully extracted project 'Muncie' to c:\GH\ras-commander\examples\example_projects\Muncie
2024-11-12 14:18:46,374 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-11-12 14:18:46,375 - ras_commander.RasPrj - INFO - Calling init_ras_project
2024-11-12 14:18:46,376 - ras_commander.RasPrj - INFO - Calling get_ras_exe
2024-11-12 14:18:46,377 - ras_commander.RasPrj - INFO - HEC-RAS executable found at default path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-11-12 14:18:46,378 - ras_commander.RasPrj - INFO - Finished get_ras_exe
2024-11-12 14:18:46,379 - ras_commander.RasPrj - INFO - Initializing global 'ras' object via init_ras_project function.
2024-11-12 14:18:46,380 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:18:46,381 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:18:46,384 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-1

Available plans in Bald Eagle Creek project:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          03  c:\GH\ras-commander\examples\example_projects\...   
2          04  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                15SEC      dss       u01                     1       g01   
1                10SEC      dss       u01                     1       g02   
2                10SEC      dss       u01                     1       g04   

  Mapping Interval                                Plan Title Program Version  \
0             5MIN                  Unsteady Multi  9-SA run            5.00   
1             5MIN            Unsteady Run with 2D 50ft Grid            5.10   
2             5MIN  Unsteady Run with 2D 50ft User n Value R            5.10   

  Run HTab Run PostProcess Run Sediment Run UNet Run

2024-11-12 14:18:46,636 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 14:18:46,640 - ras_commander.RasPrj - INFO - Finished _load_project_data
2024-11-12 14:18:46,641 - ras_commander.RasPrj - INFO - Calling get_boundary_conditions
2024-11-12 14:18:46,645 - ras_commander.RasPrj - INFO - Finished get_boundary_conditions
2024-11-12 14:18:46,645 - ras_commander.RasPrj - INFO - Initialization complete for project: Muncie
2024-11-12 14:18:46,646 - ras_commander.RasPrj - INFO - Plan entries: 5, Flow entries: 1, Unsteady entries: 1, Geometry entries: 3, Boundary conditions: 2
2024-11-12 14:18:46,647 - ras_commander.RasPrj - INFO - Geometry HDF files found: 5
2024-11-12 14:18:46,648 - ras_commander.RasPrj - INFO - Finished initialize
2024-11-12 14:18:46,649 - ras_commander.RasPlan - INFO - Finished set_geom
2024-11-12 14:18:46,649 - ras_commander.RasPlan - INFO - Calling set_geom
2024-11-12 14:18:46,650 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 

Set geometry for plan 02 in Bald Eagle Creek project
Set geometry for plan 05 in Muncie project

Example 4: Executing plans for both projects simultaneously in separate compute folders


2024-11-12 14:18:47,033 - ras_commander.RasCmdr - INFO - Copied project folder to destination: c:\GH\ras-commander\examples\example_projects\compute_bald_eagle
2024-11-12 14:18:47,034 - ras_commander.RasCmdr - INFO - Copied project folder to destination: c:\GH\ras-commander\examples\example_projects\compute_muncie
2024-11-12 14:18:47,035 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:18:47,036 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:18:47,037 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:18:47,038 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:18:47,042 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:18:47,043 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 14:18:47,043 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:18:47,045 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 14:18:47,057 - ras_commander.RasPrj - INFO - Ca

Execution results:
Plan 02 execution: Successful
Plan 05 execution: Successful

Example 5: Getting results paths
No results found for Bald Eagle Creek plan 02
No results found for Muncie plan 05

Note: The original project folders can now be edited while the compute operations are running in separate folders.




In [25]:
# wait 10 seconds
time.sleep(10)

In [26]:
# Example 14 is long, and is a notebook.

In [27]:
print("Executing 15_plan_key_operations.py:")
%run 15_plan_key_operations.py
print("\n")



2024-11-12 14:19:35,510 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 14:19:35,511 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 14:19:35,511 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 14:19:35,513 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 14:19:35,514 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 14:19:35,514 - ras_commander.RasExamples - INFO - Extracting project 'Balde Eagle Creek'
2024-11-12 14:19:35,514 - ras_commander.RasExamples - INFO - Project 'Balde Eagle Creek' already exists. Deleting existing folder...
2024-11-12 14:19:35,520 - ras_commander.RasExamples - INFO - Existing folder for project 'Balde Eagle Creek' has been deleted.
2024-11-12 14:19:35,575 - ras_

Executing 15_plan_key_operations.py:
Example 15: Plan Key Operations
------------------------------------------
Working with Plan: 01

1. Initial Plan Values:
  Computation Interval: 2MIN
  Simulation Date: 18FEB1999,0000,24FEB1999,0500
  Short Identifier: UnsteadyFlow
  UNET D1 Cores: 0

2. Updating Run Flags:
  Run flags updated.

3. Updating Plan Intervals:
  Plan intervals updated.

4. Current Plan Description:


2024-11-12 14:19:35,679 - ras_commander.RasPlan - INFO - Calling update_plan_description
2024-11-12 14:19:35,680 - ras_commander.RasPlan - INFO - Calling get_plan_path
2024-11-12 14:19:35,681 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:19:35,681 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:19:35,682 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 14:19:35,682 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:19:35,682 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:19:35,686 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:19:35,687 - ras_commander.RasPlan - INFO - Finished get_plan_path
2024-11-12 14:19:35,689 - ras_commander.RasPlan - INFO - Updated description in plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p01
2024-11-12 14:19:35,690 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-1

  

4. Updating Plan Description:
  Plan description updated.

5. Updating Simulation Date:
  Simulation date updated to: 2024-11-12 14:19:35.771581 - 2024-11-13 14:19:35.771581

6. Updated Plan Values:
  Computation Interval: 5SEC
  Simulation Date: 12NOV2024,1419,13NOV2024,1419
  Short Identifier: UnsteadyFlow
  UNET D1 Cores: 0

7. Updated Plan Description:
  This is an updated plan description for testing purposes.

Example 15 completed.




In [28]:
# wait 10 seconds
time.sleep(10)

In [29]:
print("Executing 16_scanning_ras_project_info.py:")
%run 16_scanning_ras_project_info.py
print("\n")


2024-11-12 14:19:45,886 - ras_commander.LoggingConfig - DEBUG - Calling __init__
2024-11-12 14:19:45,888 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 14:19:45,888 - ras_commander.LoggingConfig - DEBUG - Calling _load_project_data
2024-11-12 14:19:45,889 - ras_commander.LoggingConfig - DEBUG - Calling _find_zip_file
2024-11-12 14:19:45,889 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 14:19:45,890 - ras_commander.LoggingConfig - DEBUG - Finished _find_zip_file
2024-11-12 14:19:45,891 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 14:19:45,894 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 14:19:45,895 - ras_commander.LoggingConfig - DEBUG - Finished _load_project_data
2024-11-12 14:19:45,896 - ras_commander.LoggingConfig -

Executing 16_scanning_ras_project_info.py:


2024-11-12 14:19:46,139 - ras_commander.RasExamples - INFO - Successfully extracted project 'ContractionExpansionMinorLosses' to c:\GH\ras-commander\examples\example_projects\ContractionExpansionMinorLosses
2024-11-12 14:19:46,140 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-11-12 14:19:46,141 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 14:19:46,142 - ras_commander.RasExamples - INFO - Extracting project 'Culvert Hydraulics'
2024-11-12 14:19:46,180 - ras_commander.RasExamples - INFO - Successfully extracted project 'Culvert Hydraulics' to c:\GH\ras-commander\examples\example_projects\Culvert Hydraulics
2024-11-12 14:19:46,181 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-11-12 14:19:46,182 - ras_commander.RasExamples - INFO - ----- RasExamples Extracting Project -----
2024-11-12 14:19:46,183 - ras_commander.RasExamples - INFO - Extracting project 'Culverts wit

Summary for category '1D Unsteady Flow Hydraulics' has been written to: c:\GH\ras-commander\examples\ras_example_categories\ras-commander 1D Unsteady Flow Hydraulics summary.txt


2024-11-12 14:19:51,352 - ras_commander.RasExamples - INFO - Successfully extracted project 'Muncie' to c:\GH\ras-commander\examples\example_projects\Muncie
2024-11-12 14:19:51,353 - ras_commander.RasExamples - INFO - ----- RasExamples Extraction Complete -----
2024-11-12 14:19:51,355 - ras_commander.LoggingConfig - DEBUG - Finished extract_project
2024-11-12 14:19:51,360 - ras_commander.RasPrj - INFO - Calling init_ras_project
2024-11-12 14:19:51,361 - ras_commander.RasPrj - INFO - Calling get_ras_exe
2024-11-12 14:19:51,362 - ras_commander.RasPrj - INFO - HEC-RAS executable found at default path: C:\Program Files (x86)\HEC\HEC-RAS\6.6\Ras.exe
2024-11-12 14:19:51,364 - ras_commander.RasPrj - INFO - Finished get_ras_exe
2024-11-12 14:19:51,365 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:19:51,368 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:19:51,371 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:19:51,373 - ras_commander.

Summary for category '2D Unsteady Flow Hydraulics' has been written to: c:\GH\ras-commander\examples\ras_example_categories\ras-commander 2D Unsteady Flow Hydraulics summary.txt
All category summaries have been generated.
Cleaned up original extracted example projects.




In [30]:
# wait 10 seconds
time.sleep(10)

In [31]:
print("Executing xx_edge_cases.py:")
%run xx_edge_cases.py
print("\n")

2024-11-12 14:20:01,582 - ras_commander.LoggingConfig - DEBUG - Calling __init__
2024-11-12 14:20:01,584 - ras_commander.RasExamples - INFO - Example projects folder: c:\GH\ras-commander\examples\example_projects
2024-11-12 14:20:01,584 - ras_commander.LoggingConfig - DEBUG - Calling _load_project_data
2024-11-12 14:20:01,586 - ras_commander.LoggingConfig - DEBUG - Calling _find_zip_file
2024-11-12 14:20:01,586 - ras_commander.RasExamples - INFO - Found zip file: c:\GH\ras-commander\examples\Example_Projects_6_6.zip
2024-11-12 14:20:01,587 - ras_commander.LoggingConfig - DEBUG - Finished _find_zip_file
2024-11-12 14:20:01,589 - ras_commander.RasExamples - INFO - Loading project data from CSV...
2024-11-12 14:20:01,592 - ras_commander.RasExamples - INFO - Loaded 68 projects from CSV. Use list_categories() and list_projects() to explore them.
2024-11-12 14:20:01,592 - ras_commander.LoggingConfig - DEBUG - Finished _load_project_data
2024-11-12 14:20:01,593 - ras_commander.LoggingConfig -

Executing xx_edge_cases.py:
Available plans:
  plan_number                                          full_path  \
0          01  c:\GH\ras-commander\examples\example_projects\...   
1          02  c:\GH\ras-commander\examples\example_projects\...   

  Computation Interval DSS File Flow File Friction Slope Method Geom File  \
0                 2MIN      dss       u02                     2       g01   
1                 2MIN      dss       f02                     1       g01   

  Mapping Interval                     Plan Title Program Version Run HTab  \
0            1HOUR  Unsteady with Bridges and Dam            5.00        1   
1              NaN                Steady Flow Run             NaN        1   

  Run PostProcess Run Sediment Run UNet Run WQNet Short Identifier  \
0               1            0        1         0     UnsteadyFlow   
1               1          NaN        1       NaN        SteadyRun   

                   Simulation Date UNET D1 Cores UNET Use Existing IB Ta

2024-11-12 14:20:01,770 - ras_commander.RasPrj - INFO - Calling initialize
2024-11-12 14:20:01,771 - ras_commander.RasPrj - INFO - Calling find_ras_prj
2024-11-12 14:20:01,773 - ras_commander.RasPrj - INFO - Finished find_ras_prj
2024-11-12 14:20:01,773 - ras_commander.RasPrj - INFO - Calling _load_project_data
2024-11-12 14:20:01,776 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [SinglePlanTest]\BaldEagle.p01
2024-11-12 14:20:01,777 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [SinglePlanTest]\BaldEagle.p02
2024-11-12 14:20:01,781 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:20:01,784 - ras_commander.RasPrj - INFO - Found 1 geometry entries
2024-11-12 14:20:01,784 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 14:20:01,787 - ras_commander.RasPrj - INFO - Finished _load_project_data
2024-11-12 14:20:01

Execution of plan 01 completed using compute_test_mode

Example 2: Executing a single plan using compute_parallel


2024-11-12 14:20:02,171 - ras_commander.RasCmdr - ERROR - Error running plan: 01
2024-11-12 14:20:02,172 - ras_commander.RasCmdr - ERROR - Error message: 
2024-11-12 14:20:02,173 - ras_commander.RasCmdr - INFO - Total run time for plan 01: 0.06 seconds
2024-11-12 14:20:02,174 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 14:20:02,174 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:20:02,175 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:20:02,178 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p01
2024-11-12 14:20:02,180 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p02
2024-11-12 14:20:02,182 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:20:02,183 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:20:02,186 - ras_co

Parallel execution of single plan results:
Plan 01: Failed

Example 3: Executing a single plan using compute_test_mode with a string input


2024-11-12 14:20:02,454 - ras_commander.RasCmdr - ERROR - Error running plan: 02
2024-11-12 14:20:02,454 - ras_commander.RasCmdr - ERROR - Error message: 
2024-11-12 14:20:02,455 - ras_commander.RasCmdr - INFO - Total run time for plan 02: 0.06 seconds
2024-11-12 14:20:02,456 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 14:20:02,457 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:20:02,459 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:20:02,462 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [SinglePlanTestString]\BaldEagle.p01
2024-11-12 14:20:02,464 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [SinglePlanTestString]\BaldEagle.p02
2024-11-12 14:20:02,466 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:20:02,467 - ras_commander.RasPrj - INFO - Calling get

Execution of plan 02 completed using compute_test_mode with string input

Example 4: Executing a single plan using compute_parallel with a string input


2024-11-12 14:20:02,712 - ras_commander.RasUtils - INFO - Successfully updated file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p01
2024-11-12 14:20:02,714 - ras_commander.RasUtils - INFO - Finished update_file
2024-11-12 14:20:02,715 - ras_commander.RasPrj - INFO - Calling get_plan_entries
2024-11-12 14:20:02,716 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:20:02,717 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:20:02,720 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p01
2024-11-12 14:20:02,723 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek\BaldEagle.p02
2024-11-12 14:20:02,726 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:20:02,728 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:20:02,733 - ras_commander.Ras

Parallel execution of single plan (string input) results:
Plan 01: Failed

Example 5: Attempting to execute with an empty plan list


2024-11-12 14:20:03,083 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-12 14:20:03,083 - ras_commander.RasPrj - INFO - Finished check_initialized
2024-11-12 14:20:03,086 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [EmptyPlanList]\BaldEagle.p01
2024-11-12 14:20:03,088 - ras_commander.RasPrj - DEBUG - Parsed plan file: c:\GH\ras-commander\examples\example_projects\Balde Eagle Creek [EmptyPlanList]\BaldEagle.p02
2024-11-12 14:20:03,090 - ras_commander.RasPrj - INFO - Finished get_plan_entries
2024-11-12 14:20:03,090 - ras_commander.RasPrj - INFO - Calling get_geom_entries
2024-11-12 14:20:03,095 - ras_commander.RasPrj - INFO - Found 1 geometry entries
2024-11-12 14:20:03,096 - ras_commander.RasPrj - INFO - Finished get_geom_entries
2024-11-12 14:20:03,096 - ras_commander.RasPrj - INFO - Calling get_flow_entries
2024-11-12 14:20:03,097 - ras_commander.RasPrj - INFO - Calling check_initialized
2024-11-


Example 6: Attempting to execute with a non-existent plan number





# Guide for Locating Relevant Examples:

## Category: 1D Unsteady Flow Hydraulics

### Flow Hydrograph

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Culvert Hydraulics | Beav_Culvert.p01 |
| Culverts with Flap Gates | CulvertFlap.p01 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Inline Structure with Gated Spillways | Inline_3Gates.p01 |
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| JunctionHydraulics | JunctionHydraulics.p01, JunctionHydraulics.p02, JunctionHydraulics.p04, JunctionHydraulics.p03 |
| Lateral Strcuture with Gates | TwoGateGrps.p01 |
| Lateral Structure connected to a River Reach | 3Reach_with_Lat.p01 |
| Lateral Structure Overflow Weir | LatMultXS.p01 |
| Lateral Structure with Culverts | LatWeirCulverts.p01 |
| Lateral Structure with Culverts and Gates | Beav_Lateral.p01 |
| Levee Breaching | LeveeBreach.p01 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Pumping Station | Pumps.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| Rule Operations | Canal.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |
| Unsteady Flow Encroachment Analysis | UnsteadyEncroachment.p03, UnsteadyEncroachment.p02 |

### Normal Depth

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Culvert Hydraulics | Beav_Culvert.p01 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| JunctionHydraulics | JunctionHydraulics.p01, JunctionHydraulics.p02, JunctionHydraulics.p04, JunctionHydraulics.p03 |
| Lateral Structure connected to a River Reach | 3Reach_with_Lat.p01 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Rule Operations | Canal.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Gate Opening

| Example | Plan File |
|---------|-----------|
| Balde Eagle Creek | BaldEagle.p01, BaldEagle.p02 |
| ContractionExpansionMinorLosses | ContExp.p01, ContExp.p02, ContExp.p03, ContExp.p04 |
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Inline Structure with Gated Spillways | Inline_3Gates.p01 |
| Lateral Strcuture with Gates | TwoGateGrps.p01 |
| Lateral Structure with Culverts and Gates | Beav_Lateral.p01 |
| Simplified Physical Breaching | BaldEDmbrk.p01 |

### Stage Hydrograph

| Example | Plan File |
|---------|-----------|
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| Rule Operations | Canal.p01 |

### Lateral Inflow Hydrograph

| Example | Plan File |
|---------|-----------|
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| Multiple Reaches with Hydraulic Structures | 3ReachUnsteady.p01, 3ReachUnsteady.p02 |
| Pumping Station | Pumps.p01 |
| Rule Operations | Canal.p01 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Uniform Lateral Inflow Hydrograph

| Example | Plan File |
|---------|-----------|
| Dam Breaching | BaldEagleDamBrk.p07, BaldEagleDamBrk.p02, BaldEagleDamBrk.p10, BaldEagleDamBrk.p06, BaldEagleDamBrk.p08, BaldEagleDamBrk.p04, BaldEagleDamBrk.p03, BaldEagleDamBrk.p05, BaldEagleDamBrk.p09, BaldEagleDamBrk.p11 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |

### Unknown or Special Conditions

| Example | Plan File |
|---------|-----------|
| Culverts with Flap Gates | CulvertFlap.p01 |
| Elevation Controled Gates | ElevControledGates.p01 |
| Internal Stage and Flow Boundary Condition | IBStageFlowTest.p01 |
| Lateral Structure Overflow Weir | LatMultXS.p01 |
| Lateral Structure with Culverts | LatWeirCulverts.p01 |
| Levee Breaching | LeveeBreach.p01 |
| Pumping Station with Rules | PumpRule.p03 |
| UngagedAreaInflows | Ungaged_area.p03, Ungaged_area.p02 |
| Unsteady Flow Encroachment Analysis | UnsteadyEncroachment.p03, UnsteadyEncroachment.p02 |

## Category: 2D Unsteady Flow Hydraulics

# HEC-RAS Unsteady Flow Examples Analysis

## Version 6.30

### Example: Muncie

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | Muncie.p01 (Unsteady Multi  9-SA run) |
| Normal Depth             | Muncie.p03 (Unsteady Run with 2D 50ft Grid) |
|                          | Muncie.p04 (Unsteady Run with 2D 50ft User n Value R) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| Muncie  | 6.30    | - Flow Hydrograph (White River, Muncie, Station 15696.24) | 1. Muncie.p01: Unsteady Multi 9-SA run |
|         |         | - Normal Depth (White River, Muncie, Station 237.6455)    | 2. Muncie.p03: Unsteady Run with 2D 50ft Grid |
|         |         |                                                           | 3. Muncie.p04: Unsteady Run with 2D 50ft User n Value R |

## Version 6.00

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p06 (Gridded Precip - Infiltration) |
| Gate Opening             |  |
| Lateral Inflow Hydrograph |  |
| Uniform Lateral Inflow Hydrograph |  |
| Normal Depth             |  |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 6.00 | - Flow Hydrograph | 1. BaldEagleDamBrk.p06: Gridded Precip - Infiltration |
|                     |      | - Gate Opening |  |
|                     |      | - Lateral Inflow Hydrograph |  |
|                     |      | - Uniform Lateral Inflow Hydrograph |  |
|                     |      | - Normal Depth |  |

## Version 5.10

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p13 (PMF with Multi 2D Areas) |
| Gate Opening             | BaldEagleDamBrk.p15 (1d-2D Dambreak Refined Grid) |
| Lateral Inflow Hydrograph | BaldEagleDamBrk.p02 (SA to Detailed 2D Breach) |
| Uniform Lateral Inflow Hydrograph | BaldEagleDamBrk.p05 (Single 2D area with Bridges FEQ) |
| Normal Depth             | BaldEagleDamBrk.p01 (SA to Detailed 2D Breach FEQ) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 5.10 | - Flow Hydrograph | 1. BaldEagleDamBrk.p13: PMF with Multi 2D Areas |
|                     |      | - Gate Opening | 2. BaldEagleDamBrk.p15: 1d-2D Dambreak Refined Grid |
|                     |      | - Lateral Inflow Hydrograph | 3. BaldEagleDamBrk.p02: SA to Detailed 2D Breach |
|                     |      | - Uniform Lateral Inflow Hydrograph | 4. BaldEagleDamBrk.p05: Single 2D area with Bridges FEQ |
|                     |      | - Normal Depth | 5. BaldEagleDamBrk.p01: SA to Detailed 2D Breach FEQ |

## Version 5.00

### Example: BaldEagleCrkMulti2D

| Boundary Condition Types | Plan Files |
|--------------------------|------------|
| Flow Hydrograph          | BaldEagleDamBrk.p17 (2D to 1D No Dam) |
| Gate Opening             | BaldEagleDamBrk.p18 (2D to 2D Run) |
| Lateral Inflow Hydrograph | BaldEagleDamBrk.p19 (SA to 2D Dam Break Run) |
| Uniform Lateral Inflow Hydrograph | BaldEagleDamBrk.p03 (Single 2D Area - Internal Dam Structure) |
| Normal Depth             | BaldEagleDamBrk.p04 (SA to 2D Area Conn - 2D Levee Structure) |

| Project | Version | Boundary Conditions | Plans |
|---------|---------|---------------------|-------|
| BaldEagleCrkMulti2D | 5.00 | - Flow Hydrograph | 1. BaldEagleDamBrk.p17: 2D to 1D No Dam |
|                     |      | - Gate Opening | 2. BaldEagleDamBrk.p18: 2D to 2D Run |
|                     |      | - Lateral Inflow Hydrograph | 3. BaldEagleDamBrk.p19: SA to 2D Dam Break Run |
|                     |      | - Uniform Lateral Inflow Hydrograph | 4. BaldEagleDamBrk.p03: Single 2D Area - Internal Dam Structure |
|                     |      | - Normal Depth | 5. BaldEagleDamBrk.p04: SA to 2D Area Conn - 2D Levee Structure |

### Need the same tables for geometry key types

Also need to open and save all example files in 6.5 using HECRASController (making a function to do this)

This can be used to help find examples if the user wants to do something that the library is currently not capable of.  

Using the RasExamples class, the user can extract these files and inspect them to find specific examples.  